src/sequence/mod.rs

```diff
@@ -409,20 +409,55 @@ mod tests {
     use crate::{
         Error, Keypair, PublicKey, Result, Sequence, SequenceAddress, SequenceDataWriteOp,
         SequenceEntry, SequenceIndex, SequenceKind, SequencePermissions, SequencePolicyWriteOp,
-        SequencePrivatePermissions, SequencePublicPermissions, SequencePublicPolicy, SequenceUser,
+        SequencePrivatePermissions, SequencePrivatePolicy, SequencePublicPermissions,
+        SequencePublicPolicy, SequenceUser,
     };
+
     use proptest::prelude::*;
     use rand::rngs::OsRng;
     use rand::seq::SliceRandom;
     use std::collections::BTreeMap;
+    use std::sync::Arc;
     use xor_name::XorName;
 
+    fn sign_data_op(
+        mut op: SequenceDataWriteOp<SequenceEntry>,
+        keypair: &Keypair,
+    ) -> Result<SequenceDataWriteOp<SequenceEntry>> {
+        let bytes = bincode::serialize(&op.crdt_op).map_err(|_| "Could not serialize op")?;
+        let signature = keypair.sign(&bytes);
+        op.signature = Some(signature);
+        Ok(op)
+    }
+
+    fn sign_public_policy_op(
+        mut op: SequencePolicyWriteOp<SequencePublicPolicy>,
+        keypair: &Keypair,
+    ) -> Result<SequencePolicyWriteOp<SequencePublicPolicy>> {
+        let bytes = bincode::serialize(&op.crdt_op).map_err(|_| "Could not serialize op")?;
+        let signature = keypair.sign(&bytes);
+        op.signature = Some(signature);
+        Ok(op)
+    }
+
+    fn sign_private_policy_op(
+        mut op: SequencePolicyWriteOp<SequencePrivatePolicy>,
+        keypair: &Keypair,
+    ) -> Result<SequencePolicyWriteOp<SequencePrivatePolicy>> {
+        let bytes = bincode::serialize(&op.crdt_op).map_err(|_| "Could not serialize op")?;
+        let signature = keypair.sign(&bytes);
+        op.signature = Some(signature);
+        Ok(op)
+    }
+
     #[test]
     fn sequence_create_public() {
-        let actor = generate_public_key();
+        let actor1_keypair = Keypair::new_ed25519(&mut OsRng);
+        let actor = actor1_keypair.public_key();
         let sequence_name = XorName::random();
         let sequence_tag = 43_000;
-        let sequence = Sequence::new_public(actor, actor, sequence_name, sequence_tag);
+        let sequence =
+            Sequence::new_public(actor, "actor".to_string(), sequence_name, sequence_tag);
```

Comment:
Cosmetic here as it's for the test, but the actor (second arg.) could be `actor1_keypair.public_key().to_string()` instead of `"actor"`.
The first arg. shall be the PK this ops are being signed with, so perhaps change that var name to something else?